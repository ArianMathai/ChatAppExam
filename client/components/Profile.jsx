import { useContext, useEffect, useState } from "react";
import { LoginContext } from "../context/LoginContext";

function Profile() {
  const { user, email, username } = useContext(LoginContext);
  const [newBio, setNewBio] = useState("");
  const [bio, setBio] = useState("");
  const [updateSuccess, setUpdateSuccess] = useState(null);

  async function fetchBio() {
    if (!email) {
      return;
    }

    try {
      const url = new URL(`/api/login/getBio/${email}`, window.location.origin);
      const res = await fetch(url.toString());

      if (res.ok) {
        const data = await res.json();
        setBio(data.bio);
      } else {
        console.log("Could not retrieve bio");
      }
    } catch (error) {
      console.error("Error fetching bio:", error);
    }
  }

  async function handleUpdateBio() {
    const response = await fetch("/api/login/updateBio", {
      method: "PUT",
      body: JSON.stringify({ email: email, newBio: newBio }),
      headers: {
        "Content-Type": "application/json",
      },
    });

    if (response.ok) {
      const data = await response.json();
      setBio(data.bio);
      setUpdateSuccess(true);
    } else {
      setUpdateSuccess(false);
    }
  }

  useEffect(() => {
    if (email) {
      fetchBio();
    }
  }, [email]);

  return username ? (
    <div className={"profile-container"}>
      <h2 className={"profile-heading"}>Profile</h2>
      <p className={"profile-info"}>Username: {user.name}</p>
      <p className={"profile-info"}>Email: {user.email}</p>
      <div className={"profile-info"}>Bio: {bio}</div>
      <div className={"profile-info"}>
        <label className={"bio-label"} htmlFor="bio">
          Update bio:
        </label>
        <textarea
          id="bio"
          className={"bio-textarea"}
          value={newBio}
          onChange={(e) => setNewBio(e.target.value)}
        />
      </div>
      <button className={"update-bio-button"} onClick={handleUpdateBio}>
        Update Bio
      </button>
      {updateSuccess !== null && (
        <p className={"update-message"}>
          {updateSuccess
            ? "Bio updated successfully!"
            : "Failed to update bio."}
        </p>
      )}
    </div>
  ) : (
    <h2>Login to see profile</h2>
  );
}
export default Profile;
