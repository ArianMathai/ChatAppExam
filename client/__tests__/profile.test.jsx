import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import Profile from "../components/Profile";

describe("snapshot testing profile", () => {
    it("should render not logged in if not logged in", () => {
        const component = renderer.create(
            <MemoryRouter>
                <Profile />
            </MemoryRouter>,
        );
        expect(component).toMatchSnapshot();
    });

});
