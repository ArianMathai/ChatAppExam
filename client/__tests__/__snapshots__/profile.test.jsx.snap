// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing profile should render not logged in if not logged in 1`] = `
<h2>
  Login to see profile
</h2>
`;
